### ISSUE TITLE
Default Initialization of RNNs

### ISSUE THREAD

**Issue by emanjavacas** [clab/dynet#1448]
Hi there,

I am trying to find out what is the default intialization for all RNN parameters in DyNet. I have a model in PyTorch that is consistently underperforming its counterpart in DyNet and I want to find out why. I couldn't find any specific  information neither by looking at the docs nor the C++ code (which doesn't imply that isn't there, of course). I'd appreciate any hints!

**Comment by FilippoC** [comment]
Hi,

it depends of the parameter, see the calls to add_parameters here for example: https://github.com/clab/dynet/blob/master/dynet/lstm.cc#L324 
when no specific initializer is given to add_parameters, it *usually* uses the Glorot Initializer: http://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf?hc_location=ufi
(there are some special cases like https://github.com/clab/dynet/blob/master/dynet/model.cc#L333 )

**Comment by pmichel31415** [comment]
Yes @FilippoC is correct. I will leave this open as an improvement though because it should be documented better.

**Comment by emanjavacas** [comment]
Hi thanks for the info. I still can't find this initialization in the code for VanillaLSTMBuilder (same for LSTM).
The formulas you posted seem to correspond to a xavier uniform initialization, perhaps it's not a good idea to mention that in the docs as well.

**Comment by FilippoC** [comment]
@emanjavacas  I linked the initialization code of VanillaLSTM in my first post. Just check the calls to add_parameters() in the constructor of the builder you are interested in.

**Comment by emanjavacas** [comment]
Ah, sorry, first time I checked the link to lstm.cc I dind't see that if no scale is passed, the default is Glorot. I guess this means that even simple linear layers and in general all tensors would be initialized with this glorot uniform scheme unless it is manually changed.

Is this correct?

**Comment by FilippoC** [comment]
@emanjavacas yes.
